src/server/query_processor.py

```diff
@@ -14,34 +13,24 @@
     DEFAULT_FILE_SIZE_KB,
     EXAMPLE_REPOS,
     MAX_DISPLAY_SIZE,
-    templates,
 )
 from server.server_utils import Colors, log_slider_to_size
 
-if TYPE_CHECKING:
-    from fastapi import Request
-    from starlette.templating import _TemplateResponse
-
 
 async def process_query(
-    request: Request,
-    *,
     input_text: str,
     slider_position: int,
     pattern_type: str = "exclude",
     pattern: str = "",
-    is_index: bool = False,
     token: str | None = None,
-) -> _TemplateResponse:
+) -> dict[str, Any]:
```

Comment:
Let's use something else than `Any` here. At least use `object` for now.
```suggestion
) -> dict[str, object]:
```

And remove the import statement of `Any`.

Reply:
Using `object` as type for dict value leads to a lot of casting, for example in ingest.py : 
```python
            error_response = IngestErrorResponse(
                error=str(context["error"]),
                repo_url=str(context.get("repo_url", input_text)),
            )
```

I know it's not that good to have `Any` as a type,

Reply:
We can just leave it at `Any` for now if you add a `TODO: ...`

Reply:
No worry i've finally changed the function a bit to return a `Union[IngestErrorresponse, IngestSuccessResponse]`.
It's way more clean this way !